nats-base-client/protocol.ts

```diff
@@ -378,6 +378,7 @@ export class ProtocolHandler implements Dispatcher<ParserEvent> {
     sub.received += 1;
 
     if (sub.callback) {
+      sub.cancelTimeout();
```

Comment:
Not sure what semantics you want for a subscription timeout, but if you cancel here before invoking callback, basically saying we got a message, so stop the timeout timer, should you not set it again after the callback? 
In other words, the way I read it, you set the timeout in the constructor and timeout is stopped when a message is received. But the subscription will not timeout again, even if no message is received after that?